### ISSUE TITLE
[Feature]: Allow RSO enabled auction houses to have users pay for all accounts created.

### ISSUE THREAD

**Issue by GageBachik** [metaplex-foundation/metaplex-program-library#347]
### Which package is this feature request for?

auction-house

### Feature

This already exists for auction house configs where RSO is disabled. Having users pay for all accounts they need in a program and then sending that money back on the closure of said accounts seems to be the standard practice on Solana. Forcing the AH to have a fee payer account if it's the signer for transactions seems counterintuitive by default.

It would be nice to have the same functionality regardless of RSO that way we don't have to keep adding tx's to all AH calls to bring that functionality back.

### Standard Change?

No

### Ideal solution or implementation

Implement logic to take and return all rent fee's from the user who is interacting with auction house.

### Alternative solutions or implementations

The current way of doing this is to overcharge users ahead of time for account creation. 
Basically, add a tx that sends that fee to the fee payer account.
